src/services/maci/service.ts

```diff
@@ -97,12 +97,15 @@ export class MaciService {
    */
   async deployInitialVoiceCreditProxy({
     amount,
+    forceDeploy = false,
   }: IDeployInitialVoiceCreditProxyArgs): Promise<string> {
-    const address = this.storage.getAddress(
-      EContracts.ConstantInitialVoiceCreditProxy,
-      await this.getNetwork(),
-    );
-    if (address) return address;
+    if (!forceDeploy) {
+      const address = this.storage.getAddress(
+        EContracts.ConstantInitialVoiceCreditProxy,
+        await this.getNetwork(),
+      );
+      if (address) return address;
+    }
```

Comment:
Related to all cases in this file

```suggestion
    const address = this.storage.getAddress(
      EContracts.ConstantInitialVoiceCreditProxy,
      await this.getNetwork(),
    );

    if (!forceDeploy && address) {
        return address;
    }
```